### PR TITLE
Remove constants s_offset/phi_offset from 1-DoF joints

### LINKED ISSUE #2468: Modelica.Mechanics.MultiBody.Joints: remove superfluous parameters/constants
In `Modelica.Mechanics.MultiBody.Joints.Prismatic` and `Modelica.Mechanics.MultiBody.Joints.Revolute` there are superfluous parameters/constants `s_offset` and `phi_offset`, respectively.

See also #2465 for more details.

### PR DESCRIPTION
**Pull request by beutlich**
Close #2468.

### CODE REVIEW

File: ModelicaTest/MultiBody.mo [hunk 1]
```diff
@@ -441,9 +439,7 @@ for rodLength:
         n={1,0,0},
         a(fixed=false),
         s(fixed=true),
-        v(fixed=false),
-        s_offset=-0.2) annotation (Placement(transformation(extent={{12,-80},{
-                32,-60}})));
+        v(fixed=false)) annotation (Placement(transformation(extent={{12,-80},{32,-60}})));
```
Comment (tobolar) [inline ModelicaTest/MultiBody.mo]: Deleting `s_offset` here results in an offset in `j2.s` by -0.2 compared to reference results, which is fine. The consequence is that the reference results have to be generated again for further tests. This holds similarly for other test models.
Clipboard22

**Comment by beutlich** [comment]
Yes, I also wondered why we have modified constant s_offset there, if it is legal and intentional. For me it looked like a mistake of c025e058b032a9542ceba9ca0335a6a59ed6da00, but @MartinOtter claimed in https://github.com/modelica/ModelicaStandardLibrary/issues/2465#issuecomment-365172291 that is by design. After all, I am relieved that only ModelicaTest is affected, but it may affect user models.

@tobolar Should we add FixedTranslation components to the modified test models in ModelicaTest to get reproducable results?

**Comment by tobolar** [comment]
> but it may affect user models.

The referenced commit is dated on 10 Dec 2007. Which MSL Version was it? A possibility is to put a corresponding `Prismatic` joint with constant `s_offset` into "ObsoleteModelica3".

> Should we add FixedTranslation components to the modified test models in ModelicaTest to get reproducable results?

Fortunately, we just need to change position vectors of existing fixed translations. In the case of e.g. `ModelicaTest.MultiBody.FourbarVariants.SphericalAndUniversal` it is sufficient to adapt `b3(r={1.0,0,0}`.

**Comment by beutlich** [comment]
> The referenced commit is dated on 10 Dec 2007. Which MSL Version was it?

It was a rather large commit for MSL 3.0.1, see

grafik

> A possibility is to put a corresponding `Prismatic` joint with constant `s_offset` into "ObsoleteModelica3".

We could add both joints to ObsoleteModelica4, sure. But we would not have an automatic conversion for them, which basically means they will remain as zombie models forever. What do you propose? I have no strong opinion. In case of moving them to ObsoleteModelica4 we also could replace the constant by parameter.

> > Should we add FixedTranslation components to the modified test models in ModelicaTest to get reproducable results?
> 
> Fortunately, we just need to change position vectors of existing fixed translations. In the case of e.g. `ModelicaTest.MultiBody.FourbarVariants.SphericalAndUniversal` it is sufficient to adapt `b3(r={1.0,0,0}`.

OK, let me see if I get it right.

File: ModelicaTest/MultiBody.mo (hunk 1 above)
Comment (beutlich) [reply ModelicaTest/MultiBody.mo]: OK, joints are moved to yield reproducible results.

**Pull request merged**